// containers/item/index.js
import React,{Component} from 'react'
import * as R from 'ramda'
import {connect} from 'react-redux'
import {fetchItemById, addItemToBasket} from 'actions'
import {getItemById} from 'selectors'
import {Link} from 'react-router-dom'
import BasketCart from 'components/basketCard'


class Item extends Component{
  componentDidMount(){
    this.props.fetchItemById(this.props.match.params.id)
  }

  renderFields(){
    const {item} = this.props

    const columnFields = R.compose(
      R.toPairs,
      R.pick([
        'cpu',
        'camera',
        'size',
        'battery',
        'weight',
        'display',
        'memory'
      ])
    )(item)
    console.log('columnFields', columnFields)


    return columnFields.map(([key, value])=>(
      <div className='column' key={key}>
        <div className='ab-details-title'>
          <p>{key}</p>
        </div>
        <div className='ab-details-info'>
          {value}
        </div>
      </div>
    ))
  }

  renderContent(){
    const {item} = this.props
    return (
      <div className='thumbnail'>
        <div className='row'>
          <div className='col-md-6'>
                <img
                  className='img-thumbnail'
                  src={item.image}
                  alt={item.name}
                />
          </div>
                <div className='col-md-6'>
                  {this.renderFields()}
                </div>
          </div>
          <div className='caption-full'>
            <h4 className='pull-right'>${item.price}</h4>
            <h4>{item.name}</h4>
            <p>{item.description}</p>
          </div>
      </div>
    )
  }

  renderSidebar(){
    const {item, addItemToBasket} =this.props
    return (
      <div>
        <p className='lead'>Quick Shop</p>
        <BasketCart />
        <div className='form-group'>
          <h1>{item.name}</h1>
          <h2>${item.price}</h2>
        </div>
        <Link to='/' className='btn btn-info btn-block'>
            Back to store
        </Link>
        <button
          type='button'
          onClick={()=>addItemToBasket(item.id)}
          className='btn btn-success btn-block'
        >
          Add to Cart
        </button>
      </div>
    )
  }


  render(){
    const {item} = this.props
    return (
      <div className='view-container'>
        <div className='container'>
          <div className='row'>
            <div className='col-md-9'>
              {item && this.renderContent()}
            </div>
            <div className='col-md-3'>
              {item && this.renderSidebar()}
            </div>
          </div>
        </div>
      </div>
    )
  }
}
const mapStateToProps = state =>{
  return {
    item: getItemById(state, state.itemPage.id)
  }
}

const mapDispatchToProps = {
  fetchItemById,
  addItemToBasket
}

export default connect(mapStateToProps, mapDispatchToProps )(Item)
