// containers/items/index.js
import React, {Component} from 'react'
import * as R from 'ramda'
import {connect} from 'react-redux'
import {fetchItems, loadMoreItems,addItemToBasket, fetchCategories} from 'actions'
import{getItems} from 'selectors'
import {Link} from 'react-router-dom'
import Layout from 'containers/layout'


class Items extends Component {
  componentDidMount() {
    this.props.fetchItems()
    this.props.fetchCategories()
  }


  addItemToBasket(){

  }
  renderItem (item, index){
    const shortDescription =`${R.take(60, item.description)}...`
    const {addItemToBasket} =this.props
    return (
      <div className='col-sm-4 col-lg-4 col-md-4 book-list' key={index}>
          <div className='thumbnail'>
            <img
              className='img-thumbnail'
              src={item.image}
              alt={item.image}
            />
            <div className='caption'>
              <h4 className='pull-right'>${item.price}</h4>
              <h4 className=''>
                <Link to={`/items/${item.id}`}>
                  {item.name}
                </Link>
              </h4>
              <p>{shortDescription}</p>
              <p className='itemButton'>
                <button
                  className='btn btn-primary'
                  onClick={()=>addItemToBasket(item.id)}
                >
                  Buy Now
                </button>
                  <Link
                    to={`/items/${item.id}`}
                    className='btn btn-default'
                  >
                    More Info
                  </Link>
              </p>

            </div>
          </div>
        </div>
    )
}
  render() {
   const {items, loadMoreItems} = this.props
    return (
      <Layout>
        <div className='books row'>
          {items.map((item, index)=> this.renderItem(item, index))}
        </div>
        <div className='row'>
          <div className='col-md-12'>
            <button onClick={loadMoreItems}
             className='btn btn-primary pull-right'
            >
              Load More
            </button>
          </div>
        </div>
      </Layout>
    )
  }
}
const mapStateToProps = (state, ownProps) =>({
  items: getItems(state, ownProps)
})

const mapDispatchToProps = {
  fetchItems,
  loadMoreItems,
  addItemToBasket,
  fetchCategories
}

export default connect(mapStateToProps, mapDispatchToProps)(Items)
